[move reference] Add references to Sui (#16785)
MystenLabs/sui: external-crates/move/documentation/book/src/structs.md

### Abilities

As mentioned above: by default, a struct declaration is linear and ephemeral. So to allow the value

> to be used with certain operations (that copy it, drop it, store it in an
> [object](./abilities/object.md), or use it as a storage schema to define an
> [object](./abilities/object.md)), structs can be granted [abilities](./abilities.md) by annotating
> them with `has <ability>`:

Reviewer: Maybe this?

```suggestion
to be used in these ways (e.g., copied, dropped, stored in an
[object](./abilities/object.md), or used to define a storable
[object](./abilities/object.md)), structs can be granted [abilities](./abilities.md) by annotating
them with `has <ability>`:
```

```move
module a::m {
    public struct Foo has copy, drop { x: u64, y: bool }
}
```

The ability declaration can occur either before or after the struct's fields. However, only one or

> the other can be used, and not both. If declared after the struct's field, the ability declaration

Reviewer: ```suggestion
the other can be used, and not both. If declared after the struct's fields, the ability declaration
```

### Creating Structs

Values of a struct type can be created (or "packed") by indicating the struct name, followed by
value for each field.

For a struct with named fields, the order of the fields does not matter, but the field name needs to
be provided. For a struct with positional fields, the order of the fields must match the order of

> the fields in the struct definition, and instead of using `{}` to create the struct, `()` are used.

Reviewer: ```suggestion
the fields in the struct definition, and it must be created using `()` instead of `{}` to enclose the parameters.
```
